### ISSUE TITLE
What are the breaking changes upgrading from major version 9 to 11?

### ISSUE THREAD

**Issue by liangmicha** [ilyakatz/data-migrate#347]
Hi, we've recently ran into the issue: https://github.com/ilyakatz/data-migrate/issues/318 when upgrading rails to 7.2.1 from 7.1

Looking through the thread, it seems like the solution would be to upgrade to version 11, because it 9.4.4 which had the fix has been yanked. Since it's a major version upgrade, I've been looking around to see documentation around what's breaking and cannot find it. I've looked at the `ChangeLog.md` file and do not see it.

I think it's standard practice that when there is a major version change, such as from `9` to `11` there's are some breaking changes that warrants a major version change, instead of a minor version change.

Our team is now reluctant to continue using this library now because of lack of documentation.

Could someone shed some light on what are the breaking changes are so we can make a more informed decision on whether / when we'll upgrade from version 9 to version 11?

Appreciate the hard work on this library!

**Comment by vprigent** [comment]
Hi @liangmicha

As far as I am aware, version 10 was skipped as past work done ["labelled" 10](https://github.com/ilyakatz/data-migrate/blob/86a03b07a661aa73225f4852514df0382c50a364/Changelog.md#yanked-1000) didn't lead anywhere and we wanted to avoid confusion.

The main changes associated with version 11 are the support to Rails 7.2 which was pretty significant on our part.
We've since released 11.1 with additional support for multiple source of data_migrate data files.
We also dropped support for Rails 6.0 and Ruby 3.0 in 11.0

Hence the motivation behind the major version bump.
